CRM/Contribute/Form/Contribution.php

```diff
@@ -1558,8 +1558,20 @@ protected function generatePane($type, $defaults) {
    *
    * @throws \CRM_Core_Exception
    * @throws \Civi\Payment\Exception\PaymentProcessorException
+   *
+   * @deprecated since 5.68 will be removed around 5.80.
+   *
+   * Try something like
+   *  use   use \Civi\Test\FormTrait;
+   *  $form = $this->getTestForm('CRM_Contribute_Form_Contribution', $submittedValues, [
+   *   'id' =>  4;
+   *   'action' => 'update',
+   * ]);
+   * $form->processForm();
    */
   public function testSubmit($params, $action, $creditCardMode = NULL) {
+    // Note that this is really used from tests - so adding noisy deprecations would make them
+    // fail straight away.
```

Comment:
@eileenmcnaughton it's possible to test deprecated functions and still have them emit noise. e.g.:

https://github.com/civicrm/civicrm-core/blob/9df1541a8e0c1520f1879ea75cf171b13af08a91/tests/phpunit/Civi/Payment/PropertyBagTest.php#L113-L116

Reply:
@colemanw I think that's what I'm doing aren't I - first making sure it does throw an exception under normal circumstances & then suppressing it to get the value

Reply:
@eileenmcnaughton sorry I don't see where this function emits a deprecation notice. Maybe I just don't understand the flow here.

Reply:
@colemanw oh no sorry - this function doesn't - I was thinking of a different PR - no - it doesn't - but if it does then people are forced to update their test ASAP, as they would if we removed it, - so I don't think a noisy deprecation is that helpful here

Reply:
Ok, maybe that can be a "phase2" thing...